modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java

```diff
@@ -696,7 +692,10 @@ public ProcessingType getProcessingType() {
 
         @Override
         public boolean isAsync() {
```

Comment:
If there is logic in using this in the proactor is the defintion in the jdoc of the OperationInnerProcessor
"An async processor is one that eventually changes the thread where it is executing.". That would be the criteria for adding parallelism or not in the proactor.
The problem is that the isAsync in the owner class has other semantics (it indicates that the processor *would* eventually change the thread, not that it would in all cases). I'll change the name of the methods to make it clear.